<launch>

    <!-- Arguments that are passed from the top launchfile -->
    <arg name="vehicle_id" />
    <arg name="vehicle_ns" />

    <arg name="objects_ground_truth_topic_with_ns" />
    <arg name="perc_pred_obj_topic" />
    <arg name="perc_egomotion_topic" />
    <arg name="internal_communication_subns" />
    <arg name="BSM_topic" />


    <!-- Launch Nodes -->

   	<!-- Set Arguments for exact localization sensor-->
   	<arg name="sensor_name_exact_localization_sensor" value="exact_localization_sensor" />
    <node pkg="sim_sample_perception_ros_tool" type="exact_localization_sensor" name="exact_localization_sensor" output="screen">
       	<!-- Set Parameters -->
    	  <param name="vehicle_id" value="$(arg vehicle_id)" />
    	  <param name="sensor_child_frame_id_prefix" value="v$(arg vehicle_id)_$(arg sensor_name_exact_localization_sensor)_"/>

       	<!-- Remap Topics -->
        <remap from="default_egoMotion_sensor_out_topic" to="/$(arg vehicle_ns)/$(arg perc_egomotion_topic)"/>
        <remap from="default_objects_ground_truth_topic_with_ns" to="$(arg objects_ground_truth_topic_with_ns)"/>
        <remap from="default_BSM_out_topic" to="/$(arg vehicle_ns)/$(arg internal_communication_subns)/out/$(arg BSM_topic)"/>
        <remap from="default_BSM_in_topic" to="/$(arg vehicle_ns)/$(arg internal_communication_subns)/in/$(arg BSM_topic)"/>
    </node>

   	<!-- Set Arguments for abstract object sensor-->
   	<arg name="sensor_name_abstract_object_sensor" value="abstract_object_sensor" />
    <node pkg="sim_sample_perception_ros_tool" type="abstract_object_sensor" name="abstract_object_sensor" output="screen">
        <!-- Set Parameters -->
        <param name="vehicle_id" value="$(arg vehicle_id)" />
        <param name="sensor_child_frame_id_prefix" value="v$(arg vehicle_id)_$(arg sensor_name_abstract_object_sensor)_"/>

        <!-- Remap Topics -->
        <remap from="default_perceivedObjects_sensor_out_topic" to="/$(arg vehicle_ns)/$(arg perc_pred_obj_topic)"/>
        <remap from="default_objects_ground_truth_topic_with_ns" to="$(arg objects_ground_truth_topic_with_ns)"/>
    </node>

</launch>
